### PR TITLE
Macrosteps plugin first release

### PR DESCRIPTION
**Pull request by SinisterRj**

### CODE REVIEW

File: _plugins/MacroSteps.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+authors:
+- Rodrigo C. C. Silva
+license: AGPLv3
+
+# TODO
+date: 2023-01-11
+
+homepage: https://github.com/SinisterRj/OctoPrint-Macrosteps
+source: https://github.com/SinisterRj/OctoPrint-Macrosteps
+archive: https://github.com/SinisterRj/OctoPrint-Macrosteps/archive/master.zip
```
Comment (jneilliii) [inline _plugins/MacroSteps.md:16]: should be https://github.com/SinisterRj/OctoPrint-Macrosteps/archive/main.zip ?

File: _plugins/MacroSteps.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+- macro
+- gcode
+- script
+- custom
+- sidebar
+
+# TODO
+# When registering a plugin on plugins.octoprint.org, all screenshots should be uploaded not linked from external sites.
+screenshots:
+- url: /assets/img/plugins/Macrosteps/MS.jpg
```
Comment (jneilliii) [inline _plugins/MacroSteps.md:34]: plugin folder for images needs to match id. Needs to be changed to MacroSteps and all links updated.

File: _plugins/MacroSteps.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  alt: Sidebar screenshot
+  caption: Sidebar screenshot
+
+# TODO
+featuredimage: /assets/img/plugins/Macrosteps/MSfeatured.png
```
Comment (jneilliii) [inline _plugins/MacroSteps.md:39]: plugin folder for images needs to match id. Needs to be changed to MacroSteps and all links updated.

File: _plugins/MacroSteps.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+---
+
+
+# Macro Steps
+
+Tired of searching hundreds of terminal lines for the reason of a halt on your print start? Do you want more feedback on what is happening with your printer? This is the solution for your problems. This is a simple Octoprint plugin that enables you to follow the steps of your big macros in a very intuitive way. 
+
+It creates a sidepanel on Octoprint's UI with every step (or GCODE command) of your MACRO and updates it on the fly, indicating what is finished, what is running, what was skiped or what failed, helping to diagnose when something went wrong.
+
+![MacroSteps Plugin sidebar](/assets/img/plugins/onedrive_files/Macrosteps/MS.jpg)
```
Comment (jneilliii) [inline _plugins/MacroSteps.md:103]: completely wrong URL, update to your plugin's folder after renaming it properly.

**Review by jneilliii — CHANGES_REQUESTED** [review]
You need to release 0.1.0 version on the plugin's repository and then deal with the comments below.

File: _plugins/MacroSteps.md (hunk 2 above)
Comment (SinisterRj) [reply _plugins/MacroSteps.md]: Done.

File: _plugins/MacroSteps.md (hunk 3 above)
Comment (SinisterRj) [reply _plugins/MacroSteps.md]: Done.

File: _plugins/MacroSteps.md (hunk 4 above)
Comment (SinisterRj) [reply _plugins/MacroSteps.md]: Done. Sorry for my mistakes. I hope everything are right now.

**Comment by SinisterRj** [comment]
Sorry for my mistakes. I hope everything is ok now.

**Review by jneilliii — APPROVED** [review]

**Pull request merged**